File: test/fixtures/sample-sdk-python/src/workos/organizations.py
from typing import Optional, Protocol, Sequence
from workos.types.organizations import Organization
from workos.types.list_resource import ListMetadata

OrganizationsListResource = "WorkOSListResource[Organization, OrganizationListFilters, ListMetadata]"


class OrganizationsModule(Protocol):
    def list_organizations(
        self,
        *,
        domains: Optional[Sequence[str]] = None,
        limit: int = 10,
        before: Optional[str] = None,
        after: Optional[str] = None,
    ) -> "OrganizationsListResource": ...

    def get_organization(self, organization_id: str) -> Organization: ...

    def create_organization(
        self, *, name: str, domains: Optional[Sequence[str]] = None
    ) -> Organization: ...

    def delete_organization(self, organization_id: str) -> None: ...

    def _internal_helper(self) -> None: ...

File: test/fixtures/sample-sdk-php/lib/Organizations.php
<?php

namespace WorkOS;

class Organizations
{
    /**
     * List organizations.
     *
     * @param null|array $domains
     * @param int $limit
     * @param null|string $before
     * @param null|string $after
     *
     * @return array
     */
    public function listOrganizations($domains = null, $limit = 10, $before = null, $after = null)
    {
        // implementation
    }

    /**
     * Create an organization.
     *
     * @param string $name
     * @param array $domains
     * @param null|boolean $allowProfilesOutsideOrganization
     *
     * @return \WorkOS\Resource\Organization
     */
    public function createOrganization($name, $domains, $allowProfilesOutsideOrganization = null)
    {
        // implementation
    }

    /**
     * Get an organization.
     *
     * @param string $organization
     *
     * @return \WorkOS\Resource\Organization
     */
    public function getOrganization($organization)
    {
        // implementation
    }

    /**
     * Delete an organization.
     *
     * @param string $organization
     *
     * @return \WorkOS\Resource\Response
     */
    public function deleteOrganization($organization)
    {
        // implementation
    }

    private function internalHelper()
    {
        // private helper
    }
}
